jujugui/static/gui/src/app/components/added-services-list-item/test-added-services-list-item.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+var juju = {components: {}};
+var testUtils = React.addons.TestUtils;
+
+chai.config.includeStack = true;
+chai.config.truncateThreshold = 0;
+
+describe('AddedServicesListItem', function() {
+  var listItemStub;
+
+  beforeAll(function(done) {
+    // By loading this file it adds the component to the juju components.
+    YUI().use('added-services-list-item', function() { done(); });
+  });
+
+  it('renders the icon, count and display name', function() {
+    var service = {
+      getAttrs: function() {
+        return {
+          icon: 'icon.gif', unit_count: '5', name: 'demo',
+          units: {
+            toArray: function() {
+              return [];
+            }}};
+      }};
+
+    var shallowRenderer = testUtils.createRenderer();
+    shallowRenderer.render(
+        <juju.components.AddedServicesListItem
+          service={service} />);
+
+    var output = shallowRenderer.getRenderOutput();
+    assert.deepEqual(output.props.children, [
+      <img src="icon.gif" className="inspector-view__item-icon" />,
+      <span className="inspector-view__item-count">5</span>,' ' ,'demo',
+      undefined
+    ]);
```

Comment:
Not for this branch, but I almost wonder if we might want to create a react-aware set of assertion helpers that clarify this a bit better.  Again, later.